### DISCUSSION TITLE
Server Port Forwarding

### DISCUSSION THREAD

**Discussion by jojac47** [qdm12/gluetun#1426] [Q&A] [answered]
Good morning, I am running into an issue that's probably just something silly on my end. I am trying to set up port forwarding with Mulvad. I forwarded a port to the "device" registered to the gluetun. I set the variable stated in the port forwarding mullvad wiki 'FIREWALL_VPN_INPUT_PORTS= XXXXX'. I left the published ports on the gluetun container the recommended ports for the 2nd container using gluetun as its network.  I entered the forwarded port as the listening port on the 2nd containers webUI. the 2nd container is working properly with wireguard but the port forward checker on mullvad is not. 

P.S.
I saw a few discussions regarding this topic, but I didn't find an answer for my question. Would you prefer if I were to add this to the bottom of a discussion in the future?

**Comment by damajor** [comment, marked as answer]
If the service in the container binded to gluetun container use a different port than the one that is forwarded by Mullvad then you need a custom iptables rule to make the local forwarding.

Eg:

- gluetun + service A (port 1234)
- Mullvad port forward to port 4321

2 Solutions (choose 1 **_or_** 2):
First do not forget to open firewall with `FIREWALL_VPN_INPUT_PORTS=4321,1234` (both ports need to be opened)
1. Change service A to listen on 4321
2. Use a custom iptables rule to forward 4321 to 1234 as:
`iptables -t nat -A PREROUTING -i tun0 -p tcp --destination-port 4321 -j REDIRECT --to-port 1234`
Check https://github.com/qdm12/gluetun/wiki/Firewall-options#custom-iptables-rules


FYI docker/podman port mapping does not affect VPN ports, they only work locally to expose services attached to gluetun but not on the VPN side, for that purpose you have to use the solution above.
